cmd/kubectl-gadget/snapshot/snapshot.go

```diff
@@ -15,16 +15,158 @@
+	GetColumnsHeader() string
+
+	// TransformEvent is called to transform an event to columns
+	// format according to the parameters.
+	TransformEvent(*Event) string
+
+	// BuildSnapshotColsHeader returns a header with the requested custom
+	// columns that exist in the availableCols. The columns are separated by
+	// taps.
+	BuildSnapshotColsHeader([]string) string
+}
+
+// SnapshotGadget represents a gadget belonging to the snapshot category.
+type SnapshotGadget[Event SnapshotEvent] struct {
+	name        string
+	commonFlags *utils.CommonFlags
+	params      map[string]string
+	parser      SnapshotParser[Event]
+}
+
+// Run runs a SnapshotGadget and prints the output after parsing it using the
+// SnapshotParser's methods.
+func (g *SnapshotGadget[Event]) Run() error {
+	config := &utils.TraceConfig{
+		GadgetName:       g.name,
+		Operation:        "collect",
+		TraceOutputMode:  "Status",
+		TraceOutputState: "Completed",
+		CommonFlags:      g.commonFlags,
+		Parameters:       g.params,
+	}
+
+	// This callback function be called when a snapshot gadget finishes without
+	// errors and generates a list of results per node. It merges, sorts and
+	// print all of them in the requested mode.
+	callback := func(results []gadgetv1alpha1.Trace) error {
+		allEvents := []Event{}
+
+		for _, i := range results {
+			if len(i.Status.Output) == 0 {
+				continue
+			}
+
+			var events []Event
+			if err := json.Unmarshal([]byte(i.Status.Output), &events); err != nil {
+				return utils.WrapInErrUnmarshalOutput(err, i.Status.Output)
+			}
+			allEvents = append(allEvents, events...)
+		}
+
+		g.parser.SortEvents(&allEvents)
+
+		switch g.commonFlags.OutputMode {
+		case utils.OutputModeJSON:
+			b, err := json.MarshalIndent(allEvents, "", "  ")
+			if err != nil {
+				return utils.WrapInErrMarshalOutput(err)
+			}
+
+			fmt.Printf("%s\n", b)
+			return nil
+		case utils.OutputModeColumns:
+			fallthrough
+		case utils.OutputModeCustomColumns:
+			// In the snapshot gadgets it's possible to use a tabwriter because
+			// we have the full list of events to print available, hence the
+			// tablewriter is able to determine the columns width. In other
+			// gadgets we don't know the size of all columns "a priori", hence
+			// we have to do a best effort printing fixed-width columns.
+			w := tabwriter.NewWriter(os.Stdout, 0, 0, 4, ' ', 0)
+
+			fmt.Fprintln(w, g.parser.GetColumnsHeader())
+
+			for _, e := range allEvents {
+				baseEvent := e.GetBaseEvent()
+				if baseEvent.Type != eventtypes.NORMAL {
+					utils.ManageSpecialEvent(baseEvent, g.commonFlags.Verbose)
+					continue
+				}
+
+				fmt.Fprintln(w, g.parser.TransformEvent(&e))
+			}
+
+			w.Flush()
+		default:
+			return utils.WrapInErrOutputModeNotSupported(g.commonFlags.OutputMode)
+		}
+
+		return nil
+	}
+
+	return utils.RunTraceAndPrintStatusOutput(config, callback)
+}
+
+func NewSnapshotCmd() *cobra.Command {
+	SnapshotCmd := &cobra.Command{
```

Comment:
I think local variables as usually lowercase, right? (I got confused for a second thinking it was a global variable).